src/skills/AvoidBall.cpp

```diff
@@ -57,9 +57,9 @@ bt::Node::Status AvoidBall::onUpdate() {
 
     // forces from walls
     auto field = world::field->get_field();
-    double boundWidth =  field.boundary_width();
-    double halfFieldLength = field.field_length()/2;
-    double halfFieldWidth = field.field_width()/2;
+    double boundWidth =  field.get(BOUNDARY_WIDTH);
+    double halfFieldLength = field.get(FIELD_LENGTH) / 2;
+    double halfFieldWidth = field.get(FIELD_WIDTH) / 2;
```

Comment:
I don't like getters that do multiple things. I think this obfuscates behaviour. The argument that too many getters is too long, or code cloning, does not appeal to me, since the few extra lines of code add a lot of extra clarity. It is now no longer clear what you should pass into the method, I have to go look at the keys of your dictionary before I can use this function, whereas with autocomplete and specific getters and setters I can see the exact syntax already.

Reply:
- The number of LOC (lines of code) drastically decreased (in general you prefer a short code over a long code). 
- It is more easier to add new constants/values to the FieldMessage class.
- It is also more easier to review (with 20 getters you have to check if every getter returns the correct corresponding variable, someone might make an error and return the rightmost x at getLeftMostX())
- It is some kind of code cloning to have 20 getters that all have the same kind of behavior. Suppose we want to change the implementation so that it deals differently with the values than this version makes that much more easier to do.
- With this new version you can even make the FieldMessage class shorter, because it is possible to put the enums and dictionaries at different places.
- In general we want to push some of the responsibilities of the Field class downwards to the FieldMessage class. Which makes the Field class even larger. Something that we want to prevent at all cost.

I do however agree with Paul's idea to use operator overloading: int &operator[] (int) instead of having get methods. And I also agree to use an unordered_map instead of map to increase the performance.